app/services/valid_test_data_generators/applications_populater.rb

```diff
@@ -38,54 +38,49 @@ def create_participants!
-                                      headteacher_status: Application.headteacher_statuses.keys.sample,
-                                      eligible_for_funding: Faker::Boolean.boolean,
-                                      itt_provider: IttProvider.currently_approved.order("RANDOM()").first)
-
-      methods = %i[accept_application reject_application]
-
-      return if Faker::Boolean.boolean
-
-      send(methods.sample, application)
-
-      return if Faker::Boolean.boolean
-
-      application = FactoryBot.create(:application,
-                                      :pending,
-                                      :with_random_work_setting,
-                                      school:,
-                                      lead_provider:,
-                                      user:,
-                                      cohort:,
-                                      course: courses.reject { |c| c.identifier == course.identifier }.sample,
-                                      headteacher_status: Application.headteacher_statuses.keys.sample,
-                                      eligible_for_funding: Faker::Boolean.boolean,
-                                      itt_provider: IttProvider.currently_approved.order("RANDOM()").first)
-
-      return if Faker::Boolean.boolean
-
-      send(methods.sample, application)
+      return if Faker::Boolean.boolean(true_ratio: 0.3)
+
+      accept_application(application)
+
+      return if Faker::Boolean.boolean(true_ratio: 0.3)
+
+      course = courses.reject { |c| c.identifier == course.identifier }.sample
+      application = create_application(user, school, course)
+
+      return if Faker::Boolean.boolean(true_ratio: 0.3)
+
+      reject_application(application)
+    end
+
+    def create_user
+      FactoryBot.create(:user,
+                        :with_get_an_identity_id,
+                        :with_random_name,
+                        date_of_birth: Date.new(1990, 1, 1),
+                        trn: Faker::Number.unique.number(digits: 7),
+                        trn_verified: true,
+                        trn_auto_verified: true)
+    end
+
+    def create_application(user, school, course)
+      FactoryBot.create(:application,
+                        :pending,
+                        :with_random_work_setting,
+                        school:,
+                        lead_provider:,
+                        user:,
+                        cohort:,
+                        course:,
+                        headteacher_status: Application.headteacher_statuses.keys.sample,
+                        eligible_for_funding: Faker::Boolean.boolean,
+                        itt_provider: IttProvider.currently_approved.order("RANDOM()").first)
     end
 
     def accept_application(application)
-      Applications::Accept.new(application:).accept
+      Applications::Accept.new(application:, funded_place: Faker::Boolean.boolean(true_ratio: 0.7)).accept
```

Comment:
Can we only set funded place to true if:
- the cohort has funding cap set to true
- the eligible for funding is true

Otherwise we don't set the funded place if the funding cap is not set on the cohort. If it is and the eligible for funding is false we set it to false please, if it's true we can set to true/false. Happy to have a chat about this